databases/database-migrations.md

```diff
@@ -0,0 +1,94 @@
+# Database Migrations
+
+### Projected Time
+
+3 hours
+
+### Prerequisites
+
+- [Relational Databases](https://github.com/Techtonica/curriculum/blob/master/databases/relational-databases.md)
+
+### Motivation
+
+Understand how to change the schema of a database table. This includes operations to add, remove, and modify fields.
+
+Making changes to a database is often necessary as an application evolves. While some database changes require creating or deleting tables, others require modifying an existing table (called a database migration).
+
+### Objective
+
+**Apprentices will be able to:**
+- Add a field to a table
+- Remove a field from a table
+- Change the properties of a field (name, data type)
+- Change the name of a table
+- Know what database transactions are and when to apply them
+
+### Materials
+
+#### Education
+
+- [W3 Schools tutorial](https://www.w3schools.com/sql/sql_alter.asp)
+- [Khan academy video](https://www.khanacademy.org/computing/computer-programming/sql/modifying-databases-with-sql/p/altering-tables-after-creation)
+- [Transactions](https://www.postgresql.org/docs/9.6/static/tutorial-transactions.html)
+
+#### Tools
+
+- [SQL Fiddle](http://sqlfiddle.com)
+
+### Mini Lesson
+[Slides](https://docs.google.com/presentation/d/118Fg5gaMPJBQdZV8n9dcAiYLyX5BseVUcJ1fSANdm4E/edit#slide=id.g4032745045_0_3)
+
+### Common Mistakes / Misconceptions
+
+Make sure to backup your database before running a migration. If you've made a mistake in your SQL command (e.g. deleted incorrect field), reverting to a database back-up will allow you to easily get back to the previous state where your data is intact.
+
+To ensure that your migration works correctly, it is a good idea to apply it to a fake dataset first.
+
+Make sure to not slow down the database for other users during a migration. Migrations on large tables can be slow, and when migrations are in-progress, any other changes to modify the table are blocked until the migration finishes. It is a good idea to apply migrations during a low-traffic time, so it is the least disruptive to users.
+
+Include SQL commands in a transaction only when it is detrimental for some commands to succeed and others to fail. Imagine you write two SQL statements to transfer money from one bank account to another. The first statement withdraws money from one account, and the next statement deposits money to another account. They should both be wrapped in a transaction, so that it is impossible for one statement to succeed and the other to fail. Statements that include updating a user's profile picture, and updating a user last active timestamp in a database are pretty unrelated, and shouldn't be wrapped in a transaction. Using transactions when they aren't needed adds unnecessary complexity to your SQL code, and also hurts performance.
```

Comment:
This sentence was a bit confusing to read. Maybe:
```Use transactions only when it's important that separate SQL commands succeed or fail as a unit```

I'd also start by talking about when you'd want transactions in a migration (almost always, and definitely if you're adding or removing columns)

Most transaction running machinery will assume that the individual migrations are running in transactions anyway because otherwise it's impossible to resume from a failed migration.

Reply:
@rcoh thanks for the feedback. I updated the section and trimmed it down quite a bit.

It may be a good idea to break out transactions entirely -- for now I'm going to keep the section fairly concise.